onnx/onnxifi.h

```diff
@@ -0,0 +1,967 @@
+    onnxGraph graph,
+    uint32_t inputsCount,
+    const onnxTensorDescriptor* inputDescriptors,
+    uint32_t outputsCount,
+    const onnxTensorDescriptor* outputDescriptors);
+
+/**
+ * Execute operations in an ONNX graph using pre-specified locations for inputs
+ * and outputs.
+ *
+ * This function operates synchronously: it expects that graph inputs have
+ * valid values before the function is called, and will finish writing graph
+ * outputs before the function returns.
+ *
+ * The caller must successfully specify locations of input and output tensors
+ * for the graph through onnxSetGraphIO before calling this function.
+ *
+ * @param graph - graph handle created by onnxInitGraph.
+ *
+ * @retval ONNXIFI_STATUS_SUCCESS The function call succeeded and the all graph
+ *                                inputs and outputs were matched to a memory
+ *                                location.
+ * @retval ONNXIFI_STATUS_INVALID_GRAPH The function call failed because
+ *                                      graph is not an ONNX graph handle.
+ * @retval ONNXIFI_STATUS_UNIDENTIFIED_NAME The function call failed because
+ *                                          some of the ValueInfoProto.name
+ *                                          value in ModelProto.graph.input or
+ *                                          ModelProto.graph.output were not
+ *                                          specified in a call to
+ *                                          onnxSetGraphIO.
+ * @retval ONNXIFI_STATUS_NO_SYSTEM_MEMORY The function call failed because the
+ *                                         backend could not allocate enough
+ *                                         system memory to execute the model
+ *                                         graph.
+ * @retval ONNXIFI_STATUS_NO_SYSTEM_RESOURCES The function call failed due to
+ *                                            insufficient non-memory system
+ *                                            resources (e.g. file handles) to
+ *                                            execute the model graph.
+ * @retval ONNXIFI_STATUS_NO_DEVICE_MEMORY The function call failed due to
+ *                                         insufficient backend-specific memory
+ *                                         to execute the graph.
+ * @retval ONNXIFI_STATUS_NO_DEVICE_RESOURCES The function call failed due to
+ *                                            insufficient non-memory
+ *                                            backend-specific resources (e.g.
+ *                                            command queues) to execute the
+ *                                            graph.
+ * @retval ONNXIFI_STATUS_INTERNAL_ERROR The function call failed because the
+ *                                       backend experienced an unrecovered
+ *                                       internal error.
+ */
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
```

Comment:
so why some things have `onnx` prefix and some `onnxifi`. Is it intentional?

Reply:
Macros have `ONNXIFI` prefix. Functions and types have `onnx` prefix. This is because macros live in a global namespace and clash with macros in ONNX core library, but functions and types in core library are defined in their own C++ namespace, so there is no possibility for collision.

Reply:
I can make all ONNXIFI names start with `onnxifi` instead, albeit IMO this prefix is too long, often comparable to the name itself (e.g. `onnxifiStatus`)

Reply:
I guess it's fine, was just curious about the reasoning. Makes sense!